src/components/MultiHopTrade/components/MultiHopTradeConfirm/hooks/useTradeExecution.tsx

```diff
@@ -132,18 +133,37 @@ export const useTradeExecution = (
       dispatch(tradeQuoteSlice.actions.setSwapTxPending({ hopIndex, id: confirmedTradeId }))
 
       const onFail = (e: unknown) => {
-        const message = (() => {
+        const { message, translationKey, error } = (() => {
           if (e instanceof ChainAdapterError) {
-            return translate(e.metadata.translation, e.metadata.options)
+            const swapperError = swapperErrors.find(error => e.message.includes(error.value))
+
+            if (swapperError) {
+              return {
+                message: translate(swapperError.key),
+                translationKey: swapperError.key,
+                error: undefined,
+              }
+            }
+
+            return {
+              message: translate(e.metadata.translation, e.metadata.options),
+              translationKey: e.metadata.translation,
+              error: e,
+            }
+          }
+
+          return {
+            message: (e as Error).message ?? undefined,
+            translationKey: undefined,
+            error: e,
```

Comment:
suggestion:
`switch(true) {}` then `case: <pattern>.test(e.message) and default to l.155`